lib/pos_consensus_controller.go

```diff
@@ -55,9 +56,77 @@ func (cc *ConsensusController) HandleFastHostStuffBlockProposal(event *consensus
 	cc.lock.Lock()
 	defer cc.lock.Unlock()
 
-	if !consensus.IsProperlyFormedConstructVoteQCEvent(event) {
-		// If the event is not properly formed, we ignore it and log it. This should never happen.
-		return errors.Errorf("HandleFastHostStuffBlockProposal: Received improperly formed vote QC construction event: %v", event)
+	// Handle the event as a block proposal event for a regular block
+	err := cc.handleBlockProposerEvent(event, consensus.FastHotStuffEventTypeConstructVoteQC)
```

Comment:
I prefer the `if err := someFunc(); err != nil` syntax in cases like this but that a stylistic thing and I could go either way.